evadb/utils/job_scheduler.py

```diff
@@ -0,0 +1,120 @@
+        self.poll_interval_seconds = 30
+        self._evadb = evadb
+
+    def _update_next_schedule_run(self, job_catalog_entry: JobCatalogEntry) -> bool:
+        job_end_time = job_catalog_entry.end_time
+        active_status = False
+        if job_catalog_entry.repeat_interval and job_catalog_entry.repeat_interval > 0:
+            next_trigger_time = datetime.datetime.now() + datetime.timedelta(
+                seconds=job_catalog_entry.repeat_interval
+            )
+            if not job_end_time or next_trigger_time < job_end_time:
+                active_status = True
+
+        next_trigger_time = (
+            next_trigger_time if active_status else job_catalog_entry.next_scheduled_run
+        )
+        self._evadb.catalog().update_job_catalog_entry(
+            job_catalog_entry.name,
+            next_trigger_time,
+            active_status,
+        )
+        return active_status, next_trigger_time
+
+    def _get_sleep_time(self, next_job_entry: JobCatalogEntry) -> int:
+        sleep_time = self.poll_interval_seconds
+        if next_job_entry:
+            sleep_time = min(
+                sleep_time,
+                (
+                    next_job_entry.next_scheduled_run - datetime.datetime.now()
+                ).total_seconds(),
+            )
+        sleep_time = max(0, sleep_time)
+        return sleep_time
+
+    def _scan_and_execute_jobs(self):
+        while True:
+            try:
+                for next_executable_job in iter(
+                    lambda: self._evadb.catalog().get_next_executable_job(
+                        only_past_jobs=True
+                    ),
+                    None,
+                ):
+                    execution_time = datetime.datetime.now()
+
+                    # insert a job history record to mark start of this execution
+                    self._evadb.catalog().insert_job_history_catalog_entry(
+                        next_executable_job.row_id,
+                        next_executable_job.name,
+                        execution_time,
+                        None,
+                    )
+
+                    # execute the queries of the job
+                    execution_results = [
+                        execute_query(self._evadb, query)
+                        for query in next_executable_job.queries
+                    ]
+                    logger.debug(
+                        f"Exection result for job: {next_executable_job.name} results: {execution_results}"
+                    )
+
+                    # update the next trigger time for this job
+                    self._update_next_schedule_run(next_executable_job)
+
+                    # update the previosly inserted job history record with endtime
+                    self._evadb.catalog().update_job_history_end_time(
+                        next_executable_job.row_id,
+                        execution_time,
+                        datetime.datetime.now(),
+                    )
```

Comment:
We can update the result of job in the catalog. so it is queryable. We can do this in later PRs. Maybe add a TODO.